client/client_test.go

```diff
@@ -70,3 +70,30 @@ func TestParseHost(t *testing.T) {
 		}
 	}
 }
+
+func TestUpdateClientVersion(t *testing.T) {
+	cases := []struct {
+		v string
+		p string
+		q url.Values
+		e string
+	}{
+		{"1.20", "/containers/json", nil, "/v1.20/containers/json"},
+		{"v1.21", "/containers/json", url.Values{}, "/v1.21/containers/json"},
+		{"1.22", "/containers/json", url.Values{"s": []string{"c"}}, "/v1.22/containers/json?s=c"},
+		{"v1.22", "/containers/json", url.Values{"s": []string{"c"}}, "/v1.22/containers/json?s=c"},
+	}
+
+	apiClient, err := NewClient("unix:///var/run/docker.sock", "", nil, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, cs := range cases {
+		apiClient.UpdateClientVersion(cs.v)
+		g := apiClient.getAPIPath(cs.p, cs.q)
```

Comment:
My concern with checking getAPIPath instead of actually checking what the server sees is that if someone adds a new code path to open a connection then we'll miss it.  Might not be a big deal but I'd like to hear what @vdemeester thinks.
See https://github.com/docker/engine-api/blob/master/client/container_create_test.go for an example of checking it from the server side.